### DISCUSSION TITLE
Как получить значение по условию из списка вложенных массивов?

### DISCUSSION THREAD

**Discussion by erpic-ru** [totumonline/totum-ru-issues-and-discussions#116] [Как это сделать?] [answered]
Пытаюсь интегрироваться с Почтой России через SOAP. Мне приходит такой список вложенных массивов с данными по отправлению:

<img width="513" alt="Снимок экрана 2022-04-22 в 09 52 34" src="https://user-images.githubusercontent.com/91674114/164623365-170560d7-b076-4fab-a07e-d6d30b56b95a.png">

Нужно вытащить из него только дату вручения.

Подскажите, пожалуйста, при помощи каких функций можно получить значение OperDate, где Opertype [id] = 2?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
**Если мы ищем дату по `id`**

Вот у нас структура:

<img width="943" alt="Снимок экрана 2022-04-22 в 15 12 02" src="https://user-images.githubusercontent.com/55755565/164711989-272470ac-461d-42ec-aaa4-a721d333754a.png">

И мы в нее закапываемся:

```
= : #h_json[[firstLevel_2]][[secondLevel_Date]][$single]

single: $key[0]
	key: listSearch(list: #h_json[[firstLevel_2]][[secondLevel_3]][[id]]; key: "value" = 11) 

```

Это один из вариантов, можно и в другой комбинации это проделать.

**Reply by ttm-alexeypolunin** [reply]
Тут еще может возникнуть вопрос, а как эту дату преобразовать в дату Тотума, на это чуть позже отвечу.

**Reply by erpic-ru** [reply]
Спасибо большое! Через поиск веселей искать. Дату я просто подрезал по длине и Тотум ее признал.

**Reply by ttm-alexeypolunin** [reply]
Лайк! Я тож решил, что самый тотум-способ — это отрезать по длине 🧨💥🔨
